pkg/pgmodel/end_to_end_tests/drop_test.go

```diff
@@ -332,53 +333,131 @@ func TestSQLDropMetricChunk(t *testing.T) {
+			if err != nil {
+				t.Error(loc, err)
+			}
+			if count != 2 {
+				t.Errorf("unexpected row count: %v @ %v", count, loc)
+			}
 
-		err = db.QueryRow(context.Background(), `SELECT count(*) FROM _prom_catalog.series`).Scan(&count)
-		if err != nil {
-			t.Error(err)
-		}
+			err = db.QueryRow(context.Background(), `SELECT count(*) FROM _prom_catalog.series`).Scan(&count)
+			if err != nil {
+				t.Error(loc, err)
+			}
 
-		if count != 2 {
-			t.Errorf("unexpected series count: %v", count)
+			// none of the series should be removed yet
+			if count != 3 {
+				t.Errorf("unexpected series count: %v @ %v", count, loc)
+			}
+
+			err = db.QueryRow(context.Background(), `SELECT count(*) FROM _prom_catalog.series WHERE delete_epoch IS NOT NULL`).Scan(&count)
+			if err != nil {
+				t.Error(loc, err)
+			}
+
+			// one of the series should be marked for deletion
+			if count != 1 {
+				t.Errorf("unexpected series count: %v @ %v", count, loc)
+			}
+
+			err = db.QueryRow(context.Background(), `SELECT count(*) FROM _prom_catalog.label where key='name1'`).Scan(&count)
+			if err != nil {
+				t.Error(loc, err)
+			}
+
+			// none of the labels should deleted yet
+			if count != 3 {
+				t.Errorf("unexpected labels count: %v @ %v", count, loc)
+			}
 		}
 
-		err = db.QueryRow(context.Background(), `SELECT count(*) FROM _prom_catalog.label where key='name1'`).Scan(&count)
+		before_delete_correct("after first")
+
+		//rerun again -- nothing changes
+		_, err = db.Exec(context.Background(), "CALL _prom_catalog.drop_metric_chunks($1, $2)", "test", chunkEnds.Add(time.Second*5))
 		if err != nil {
-			t.Error(err)
+			t.Fatal(err)
 		}
+		before_delete_correct("after first repeat")
 
-		if count != 2 {
-			t.Errorf("unexpected labels count: %v", count)
+		// reruns don't change anything until the dead series are actually dropped
+		for i := 0; i < 6; i++ {
+			drop := fmt.Sprintf("CALL _prom_catalog.drop_metric_chunks($1, $2, now()+'%v hours')", i)
+			_, err = db.Exec(context.Background(), drop, "test", chunkEnds.Add(time.Second*5))
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			before_delete_correct(fmt.Sprintf("after loop %v", i))
 		}
 
-		//rerun again -- nothing dropped
-		err = db.QueryRow(context.Background(), "SELECT _prom_catalog.drop_metric_chunks($1, $2)", "test", chunkEnds.Add(time.Second*5)).Scan(&wasDropped)
-		if err != nil {
-			t.Fatal(err)
+		before_delete_correct("after all loops")
+
+		after_delete_correct := func(loc string) {
+			count := 0
+			err = db.QueryRow(context.Background(), `SELECT count(*) FROM prom_data.test`).Scan(&count)
+			if err != nil {
+				t.Error(loc, err)
+			}
+			if count != 2 {
+				t.Errorf("unexpected row count: %v @ %v", count, loc)
+			}
+
+			err = db.QueryRow(context.Background(), `SELECT count(*) FROM _prom_catalog.series`).Scan(&count)
+			if err != nil {
+				t.Error(loc, err)
+			}
+
+			// one of the series should be removed
```

Comment:
how is this working after the first call to drop_metric_chunks. I would of expected at least 2 calls necessary before this is true

Reply:
@cevian `drop_metric_chunks()` is called in a loop 6 times [here](https://github.com/timescale/promscale/pull/304/files#diff-59a0f996197a51fe1276a3e186e4ab912e5dc3673ee9bf2c0699549834cb1605R397-R404) before this function is ever called